File: config/config-sbc2ha/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/sensor/analog/AnalogSensorConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.sensor.analog;

import com.dfi.sbc2ha.config.sbc2ha.definition.enums.PlatformType;
import com.dfi.sbc2ha.config.sbc2ha.definition.sensor.ScheduledSensorConfig;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AnalogSensorConfig extends ScheduledSensorConfig {


    int analog;

    public AnalogSensorConfig() {
        super();
        platform = PlatformType.ANALOG;
    }
}

File: config/editor/src/app/definition/sensor/analog/AnalogSensorConfig.ts
import {ScheduledSensorConfig} from "../ScheduledSensorConfig";
import {PlatformType} from "../../enums/PlatformType";

export class AnalogSensorConfig extends ScheduledSensorConfig {
  analog: number;

  constructor(name: string, analog: number) {
    super(PlatformType.ANALOG, name);
    this.analog = analog;
  }
}
